### ISSUE TITLE
⚠️Broken Ore Dictionary Registrations (with SpongeForge)

### ISSUE THREAD

**Issue by ghost** [Alex-the-666/Ice_and_Fire#1575]
### **First off: A few details for @Alex-the-666** 

- Minecraft Version: 1.12.2
- Minecraft Forge Version: 1.12.2-14.23.5.2838 
- SpongeForge Version: 1.12.2-2825-7.1.6-RC3716
- Java Version: 1.8.0

### **Next up: A detailed explanation of the bug I am reporting:**
Before people jump to the conclusion of this being a "SpongeForge issue"- this is not and is 100% an issue with the Ice & Fire mod.

Essentially what Ice and Fire is doing is adding multiple items which aren't known to the game registry. (`type: class com.github.alexthe666.iceandfire.item.ItemGeneric`) This dictionary item (ItemGeneric) can only support one value when it comes to registering with ores alike.

The easiest way to fix this is to register your Ore Dictionary Entries after the GameRegistry.
Currently, the game denies registration of these items to prevent crashes and other issues that could occur.

The item(s) affected include (but are not limited to):
ingotSilver, nuggetSilver, oreSilver, blockSilver, gemSapphire, oreSapphire, blockSapphire, boneWither, boneWithered, boneDragon, seaSerpentScales, listAllEgg, objectEgg, bakingEgg, egg, ingredientEgg, foodSimpleEgg.

As well as this: Smelting Recipe Registration is also performed incorrectly. You can see the log below.
```
19:33:39] [Server thread/ERROR] [Sponge]: Invalid smelting recipe registration!
The mod "iceandfire" is registering a smelting recipe wrongly which causes the resulting recipe to be invalid.
A common cause which makes smelting recipes invalid occurs when registering those recipes before the used blocks are
registered, it's recommend to register them when the items are finished registering.

java.lang.IllegalStateException: Invalid smelting recipe registered by the mod "iceandfire"!
	at net.minecraft.item.crafting.FurnaceRecipes.handler$onAddSmeltingRecipe$zip000(FurnaceRecipes.java:655) [akp.class:?]
	at net.minecraft.item.crafting.FurnaceRecipes.addSmeltingRecipe(FurnaceRecipes.java:116) [akp.class:?]
	at net.minecraft.item.crafting.FurnaceRecipes.addSmelting(FurnaceRecipes.java:111) [akp.class:?]
	at net.minecraft.item.crafting.FurnaceRecipes.addSmeltingRecipeForBlock(FurnaceRecipes.java:106) [akp.class:?]
	at net.minecraftforge.fml.common.registry.GameRegistry.addSmelting(GameRegistry.java:216) [GameRegistry.class:?]
	at com.github.alexthe666.iceandfire.core.ModRecipes.preInit(ModRecipes.java:110) [ModRecipes.class:?]
	at com.github.alexthe666.iceandfire.IceAndFire.preInit(IceAndFire.java:86) [IceAndFire.class:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_181]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_181]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_181]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_181]
	at net.minecraftforge.fml.common.FMLModContainer.handleModStateEvent(FMLModContainer.java:637) [FMLModContainer.class:?]
	at sun.reflect.GeneratedMethodAccessor10.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_181]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_181]
	at com.google.common.eventbus.Subscriber.invokeSubscriberMethod(Subscriber.java:91) [minecraft_server.1.12.2.jar:?]
	at com.google.common.eventbus.Subscriber$SynchronizedSubscriber.invokeSubscriberMethod(Subscriber.java:150) [minecraft_server.1.12.2.jar:?]
	at com.google.common.eventbus.Subscriber$1.run(Subscriber.java:76) [minecraft_server.1.12.2.jar:?]
	at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:399) [minecraft_server.1.12.2.jar:?]
	at com.google.common.eventbus.Subscriber.dispatchEvent(Subscriber.java:71) [minecraft_server.1.12.2.jar:?]
	at com.google.common.eventbus.Dispatcher$PerThreadQueuedDispatcher.dispatch(Dispatcher.java:116) [minecraft_server.1.12.2.jar:?]
	at com.google.common.eventbus.EventBus.post(EventBus.java:217) [minecraft_server.1.12.2.jar:?]
	at net.minecraftforge.fml.common.LoadController.sendEventToModContainer(LoadController.java:219) [LoadController.class:?]
	at net.minecraftforge.fml.common.LoadController.propogateStateMessage(LoadController.java:197) [LoadController.class:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_181]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_181]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_181]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_181]
	at com.google.common.eventbus.Subscriber.invokeSubscriberMethod(Subscriber.java:91) [minecraft_server.1.12.2.jar:?]
	at com.google.common.eventbus.Subscriber$SynchronizedSubscriber.invokeSubscriberMethod(Subscriber.java:150) [minecraft_server.1.12.2.jar:?]
	at com.google.common.eventbus.Subscriber$1.run(Subscriber.java:76) [minecraft_server.1.12.2.jar:?]
	at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:399) [minecraft_server.1.12.2.jar:?]
	at com.google.common.eventbus.Subscriber.dispatchEvent(Subscriber.java:71) [minecraft_server.1.12.2.jar:?]
	at com.google.common.eventbus.Dispatcher$PerThreadQueuedDispatcher.dispatch(Dispatcher.java:116) [minecraft_server.1.12.2.jar:?]
	at com.google.common.eventbus.EventBus.post(EventBus.java:217) [minecraft_server.1.12.2.jar:?]
	at net.minecraftforge.fml.common.LoadController.redirect$onPost$zza000(LoadController.java:553) [LoadController.class:?]
	at net.minecraftforge.fml.common.LoadController.distributeStateMessage(LoadController.java:136) [LoadController.class:?]
	at net.minecraftforge.fml.common.Loader.preinitializeMods(Loader.java:627) [Loader.class:?]
	at net.minecraftforge.fml.server.FMLServerHandler.beginServerLoading(FMLServerHandler.java:99) [FMLServerHandler.class:?]
	at net.minecraftforge.fml.common.FMLCommonHandler.onServerStart(FMLCommonHandler.java:333) [FMLCommonHandler.class:?]
	at net.minecraft.server.dedicated.DedicatedServer.init(DedicatedServer.java:125) [nz.class:?]
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:486) [MinecraftServer.class:?]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_181]
[19:33:39] [Server thread/INFO] [FML]: Ignored smelting recipe with conflicting input: 1xtile.air@32767 = 1xtile.air@0
```

**Comment by ghost** [comment]
Extended Ore Dictionaries:
https://pastebin.com/p45WfZDn

**Comment by AlexModGuy** [comment]
fixed in 1.8.0
